### ISSUE TITLE
Cannot install latest WSL version from Microsoft Store or Github repo

### ISSUE THREAD

**Issue by jmihalich** [microsoft/WSL#10640]
### Windows Version

Microsoft Windows [Version 10.0.22621.2428]

### WSL Version

does not work

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.10.102.1-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Wed Mar 2 00:30:59 UTC 2022

### Distro Version

Ubuntu 22.04

### Repro Steps

wsl --update

or 

msiexec.exe /i c:\Users\joe\Desktop\wsl.2.0.4.0.x64.msi /L*V c:\Users\joe\Desktop\wsllogs.txt

### Expected Behavior

WSL Updates to the latest version from the microsoft store (or the github repo)

### Actual Behavior

Receive various errors at the end of the install or WSL is just dead altogether and gives me a message that says, WSL is currently upgrading.

It's completely unclear what the upgrade steps are if you have the original windows built in WSL/WSLG installed via the control panel/programs app.

But, let's keep this simple.  I have reset the system to have WSL installed from the "turn windows featurers on or off" in the control panel, and ran windows update, and it installed the latest windows update for the built in WSL.  At this point i have a functioning Ubuntu 22.04 distro working.

I downloaded the latest 2.0.4.0 distro from the github repo releases area (the  msi) and ran the command:
msiexec.exe /i c:\Users\joe\Desktop\wsl.2.0.4.0.x64.msi /L*V c:\Users\joe\Desktop\wsllogs.txt

That ran and failed with the error shown below in the screen shot.  The error is bogus because the service actually installed as i can see it in the Services control panel app, though it won't start (more likely the reason for the error).   When I click cancel it uninstalls everything cuz it rolled back.

I've attached the wsl installer log file.  I have no clue what is going on.  This has to be a bug somewhere.

I am on a standalone windows 11 box at home and my user account is an administrator account.

### Diagnostic Logs

[wsllogs.txt](https://github.com/microsoft/WSL/files/12981607/wsllogs.txt)

<img width="744" alt="wsl-install-error" src="https://github.com/microsoft/WSL/assets/15250728/f81a6a2e-02a0-4f0a-a4c9-10ce2753421d">

**Comment by jmihalich** [comment]
furthermore, if i try to install it via the Microsoft.WSL_2.0.4.0_x64_ARM64.msixbundle installer, i get the following error.

<img width="714" alt="image" src="https://github.com/microsoft/WSL/assets/15250728/a38bd97d-2a85-4e14-a3e3-70f9f734151f">

**Comment by jmihalich** [comment]
furthermore, if i try to run wsl --update from an elevated powershell terminal, the following occurs in the screenshot below.  Note that the latest version did install cuz it shows up in settings/apps as 2.0.4.0.


<img width="1374" alt="image" src="https://github.com/microsoft/WSL/assets/15250728/c3fc6d4d-00f9-4bce-9c11-8aa5cdf7b78b">

**Comment by jmihalich** [comment]
also, after the wsl --update command, if I try to start the WSLService service  in Services, i get the following error:

<img width="1268" alt="image" src="https://github.com/microsoft/WSL/assets/15250728/61fd78f9-f9e1-49e4-9737-a9ed147684c2">

**Comment by jmihalich** [comment]
Note that all of the previous update attempts were done with the system reset to the same place.  That is, with the built in wsl and default windows update to the latest built in version of WSL.  From there, I ran each of the install/update attempts.  

I made sure that any remnants of failed installs were uninstalled/removed from the Windows 11 Settings | Apps | Installed Apps page.

**Comment by jmihalich** [comment]
FYI I repeated all the above steps with 2.0.6.0, and had the exact same issues with each.

**Comment by OneBlue** [comment]
Thank you @jmihalich. It looks like you might have WSL is a weird state. Can you try to install `2.0.6` via the [MSI package](https://github.com/microsoft/WSL/releases/download/2.0.6/wsl.2.0.6.0.x64.msi) but calling msiexec from an elevated shell, or opening it directly from the explorer ? This should solve the privilege issue.

**Comment by jmihalich** [comment]
Hi, I have already done that, and it failed.  here is the screenshot.

<img width="1335" alt="image" src="https://github.com/microsoft/WSL/assets/15250728/29382fd0-1ce0-4fbb-b747-0b6b0e354cd5">

**Comment by jmihalich** [comment]
The issue with the msi installer seems to be with attempting to uninstall/install/update the WSLService windows service.

**Comment by jmihalich** [comment]
Hi, so I took a bold step, and fixed the problem.  Since the issue seemed to be related to an apparently dead service WslService, i ran the following command to uninstall it:

sc delete WslService

Note that I had previously completely uninstalled all WSL Updates and the WSL Built in feature from Windows.

Once i removed the service, I reran the MSI installer for 2.0.6.0 and that succeeded.  Now when I run wsl.exe --version, i correctly get:

`WSL version: 2.0.6.0
Kernel version: 5.15.133.1-1
WSLg version: 1.0.59
MSRDC version: 1.2.4677
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25880.1000-230602-1350.main
Windows version: 10.0.22621.2428`

**Comment by OneBlue** [comment]
I'm curious on how your machine got into that state, but since the issue is resolved I'll close this. Please reopen if you hit the same issue again.

**Comment by cgimenes** [comment]
I'm having the same issue. But `sc delete WslService` didn't work for me.
I haven't used WSL for a couple weeks and today it just crashed with:
```Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x8007019e
Error: 0x8007019e The Windows Subsystem for Linux has not been enabled.

Press any key to continue...```
Tried disabling and enabling WSL in `optionalfeatures.exe` but it didn't work.
I also tried many `wsl --update` options, but none worked.
And lastly, I tried running the .msi package, but it failed with the service error above.
I can't see the service in `services.msc`, so, it may be a different issue.
